spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -245,17 +246,19 @@ def __init__(self, resfile_path=None, clufile_path=None, folder_path=None, keep_
                     resfile_path = '{}/{}.res.{}'.format(folder_path,SORTING_NAME,shank_id)
                     clufile_path = '{}/{}.clu.{}'.format(folder_path,SORTING_NAME,shank_id)
                     
-                    # doesn't recognize the items in the sorting as being NeuroscopeSortingExtractors
-                    # all_shanks_list_se.append(NeuroscopeSortingExtractor.__init__(self, resfile_path=resfile_path,
-                    #                                                                     clufile_path=clufile_path,
-                    #                                                                     keep_mua_units=keep_mua_units))
+                    
+                    # when called with either NeuroscopeSortingExtractor(MultiSortingExtractor) or NeuroscopeSortingExtractor(SortingExtractor)
+                    # doesn't recognize the items in the sorting as being NeuroscopeSortingExtractors, thinks they are 'Nonetype'
+                    all_shanks_list_se.append(NeuroscopeSortingExtractor.__init__(self, resfile_path=resfile_path,
+                                                                                        clufile_path=clufile_path,
+                                                                                        keep_mua_units=keep_mua_units))
                     
                     # runs correct, but final object does not inherit basic functions like 'get_unit_ids' from MultiSortingExtractor
                     # and also doesn't seem to set the interval values of _all_unit_ids and such correctly...
                     # Maybe the issue lies in calling NeuroscopeSortingExtractor() within its own __init__?
-                    all_shanks_list_se.append(NeuroscopeSortingExtractor(resfile_path=resfile_path,
-                                                                         clufile_path=clufile_path,
-                                                                         keep_mua_units=keep_mua_units))
+                    # all_shanks_list_se.append(NeuroscopeSortingExtractor(resfile_path=resfile_path,
+                    #                                                      clufile_path=clufile_path,
+                    #                                                      keep_mua_units=keep_mua_units))
                     
                     
                 MultiSortingExtractor.__init__(self, all_shanks_list_se)
```

Comment:
Maybe you can make the `NeuroscopeSortingExtractor` inherit from the `MultiSortingExtractor` object? Then, if it auto-detects a single .res .clue pair, it can be a `MultiSortingExtractor` with a single `SortingExtractor` (which is fine). Otherwise, it can naturally be a `MultiSortingExtractor` with multiple `SortingExtractors`. Part of the challenge is that we currently don't have a mechanism for dealing with multi-segment recordings in the original `SortingExtractor`, which makes this kinda file forma difficult. Thoughts on this potential solution @alejoe91 @samuelgarcia ?

Reply:
To the best of my understanding, defining the entire `NeuroscopeSortingExtractor` as a subclass of `MultiSortingExtractor` leads to a chicken and egg problem, where the `MultiSortingExtractor` requires, as its `_sorting` internal or `sorting` argument, a distinct type of sorting extractor with its own defined functions such as `get_unit_ids `and `get_traces`.

Perhaps the solution would be to have a `NeuroscopeSortingExtractor(SortingExtractor)` for single shank data, and a `NeuroscopeMultiSortingExtractor(MultiSortingExtractor)` for multi-shank, which uses the `NeuroscopeSortingExtractor` as the filling sorting type?

Reply:
These seems like a good solution for now. We are thinking about supporting multiple segments (for multi-shank) in future versions, but that is still being discussed.